text/0097-log-data-model.md

```diff
@@ -0,0 +1,968 @@
+    <td>attributes["net.peer.ip"]</td>
+  </tr>
+  <tr>
+    <td>%A</td>
+    <td>string</td>
+    <td>Server IP</td>
+    <td>attributes["net.host.ip"]</td>
+  </tr>
+  <tr>
+    <td>%h</td>
+    <td>string</td>
+    <td>Remote hostname. </td>
+    <td>attributes["net.peer.name"]</td>
+  </tr>
+  <tr>
+    <td>%m</td>
+    <td>string</td>
+    <td>The request method.</td>
+    <td>attributes["http.method"]</td>
+  </tr>
+  <tr>
+    <td>%v,%p,%U,%q</td>
+    <td>string</td>
+    <td>Multiple fields that can be composed into URL.</td>
+    <td>attributes["http.url"]</td>
+  </tr>
+  <tr>
+    <td>%>s</td>
+    <td>string</td>
+    <td>Response status.</td>
+    <td>attributes["http.status_code"]</td>
+  </tr>
+  <tr>
+    <td>All other fields</td>
+    <td>any</td>
+    <td>Structured data.</td>
+    <td>attributes, key=TBD</td>
+  </tr>
+</table>
+
+
+### CloudTrail Log Event
+
+<table>
+  <tr>
+    <td>Field</td>
+    <td>Type</td>
+    <td>Description</td>
+    <td>Maps to Unified Model Field</td>
+  </tr>
+  <tr>
+    <td>eventTime</td>
+    <td>string</td>
+    <td>The date and time the request was made, in coordinated universal time (UTC).</td>
+    <td>timestamp</td>
+  </tr>
+  <tr>
+    <td>eventSource</td>
+    <td>string</td>
+    <td>The service that the request was made to. This name is typically a short form of the service name without spaces plus .amazonaws.com.</td>
+    <td>resource["service.name"]?</td>
+  </tr>
+  <tr>
+    <td>awsRegion</td>
+    <td>string</td>
+    <td>The AWS region that the request was made to, such as us-east-2.</td>
+    <td>resource["cloud.region"]</td>
+  </tr>
+  <tr>
+    <td>sourceIPAddress</td>
+    <td>string</td>
+    <td>The IP address that the request was made from.</td>
+    <td>resource["net.peer.ip"] or resource["net.host.ip"]? TBD</td>
+  </tr>
+  <tr>
+    <td>errorCode</td>
+    <td>string</td>
+    <td>The AWS service error if the request returns an error.</td>
+    <td>short_description</td>
+  </tr>
+  <tr>
+    <td>errorMessage</td>
+    <td>string</td>
+    <td>If the request returns an error, the description of the error.</td>
+    <td>body</td>
+  </tr>
+  <tr>
+    <td>All other fields</td>
+    <td>*</td>
+    <td></td>
+    <td>attributes["cloudtrail.*"]</td>
+  </tr>
+</table>
+
+
+## Appendix B: `severity_number` example mappings.
+
+|Syslog       |WinEvtLog  |Log4j |Zap   |severity_number|
+|-------------|-----------|------|------|---------------|
+|             |           |TRACE |      |TRACE          |
+|Debug        |Verbose    |DEBUG |Debug |DEBUG          |
+|Informational|Information|INFO  |Info  |INFO           |
+|Notice       |           |      |      |INFO2          |
+|Warning      |Warning    |WARN  |Warn  |WARN           |
+|Error        |Error      |ERROR |Error |ERROR          |
+|Critical     |Critical   |      |Dpanic|ERROR2         |
+|             |           |      |Panic |ERROR3         |
+|Alert        |           |FATAL |Fatal |FATAL          |
```

Comment:
java.util.logging suggestion:
ALL -> TRACE
FINEST -> DEBUG
FINER -> DEBUG2
FINE -> DEBUG3
INFO -> INFO
WARNING -> WARN
SEVERE -> ERROR

Reply:
I will add java.util.logging to the example.

Is "ALL" an actual level at which a log record can be emitted? I believe it is a value at which you can set the log filtering levels but you cannot actually generate a log message of that level.

Reply:
Added java.util.logging.

Reply:
Hey @tigrannajaryan, in case it helps, here's the mapping that we are using currently in auto-instr-java: [Config.java#L119](https://github.com/open-telemetry/opentelemetry-auto-instr-java/blob/v0.2.2/agent-bootstrap/src/main/java/io/opentelemetry/auto/config/Config.java#L119) (agree that you don't need to map `ALL` or `OFF`)

Reply:
Sorry, misunderstood the javadoc. 👍

Reply:
For completeness sake - I was able to generate log messages at both `ALL` and `OFF`:
```java
    public static void main(String[] args) {
        Logger logger = Logger.getGlobal();
        for (Handler handler: logger.getParent().getHandlers()) {
            handler.setLevel(Level.ALL);
        }
        logger.setLevel(Level.ALL);
        logger.log(Level.ALL, "this was logged at ALL level");
        logger.log(Level.OFF, "this was logged at OFF level");
    }
```
log:
```
Apr 29, 2020 8:22:13 AM Main main
ALL: this was logged at ALL level
Apr 29, 2020 8:22:14 AM Main main
OFF: this was logged at OFF level
```

Reply:
Despite that, since java.util.logging levels are ultimately numeric, (INFO = 800), and the user can subclass Level for their own levels, java.util.logging will probably use `SeverityNumber`. But that's ultimately an implementation detail that doesn't have to be answered here.